Chatkit/Auth/TokenProvider.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  TokenProvider.swift
+//  Chatkit
+//
+//  Created by Mike Pye on 06/12/2019.
+//  Copyright © 2019 Pusher Ltd. All rights reserved.
+//
+// STUB FILE. This file demonstrates the desired interface, but contains to implementation
+// TODO: Move many of these types to PusherPlatform
+
+import Foundation
+import PusherPlatform
+
+// TODO: Temporary
+extension String: Error { }
+
+// Re-export so that PusherPlatform import is not required in client code
+public typealias TokenProvider = PPTokenProvider
+public typealias TokenProviderResult = PPTokenProviderResult
+
+/// A factory for `TokenProvider`s.
+///
+/// The methods on this object create token provider instances which can interface with an HTTP endpoint in your backend, or the Chatkit Test Token Provider (for use during development)
```

Comment:
The line length should no more that a half of the current length. It's not fitting GH view port and even if you have a massive screen it's hard for eyes to follow such long line and quickly jump to the next line (without readjusting or the risk to miss a line).

Reply:
This is a problem I'm having with XCode. It renders doc comments in variable width font, making it almost impossible to judge reasonable line lengths, and it doesn't auto-wrap at a reasonable length either.

I'll dig in the settings and see if I can have something reasonable enforced.